engine/src/main/java/org/terasology/world/chunks/Chunks.java

```diff
@@ -0,0 +1,306 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.chunks;
+
+import org.joml.Math;
+import org.joml.RoundingMode;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.module.sandbox.API;
+import org.terasology.world.block.BlockRegion;
+import org.terasology.world.block.BlockRegions;
+
+@API
+public final class Chunks {
+    public static final int SIZE_X = 32;
+    public static final int SIZE_Y = 64;
+    public static final int SIZE_Z = 32;
+
+    public static final int INNER_CHUNK_POS_FILTER_X = Integer.highestOneBit(SIZE_X) - 1;
+    public static final int INNER_CHUNK_POS_FILTER_Y = Integer.highestOneBit(SIZE_Y) - 1;
+    public static final int INNER_CHUNK_POS_FILTER_Z = Integer.highestOneBit(SIZE_Z) - 1;
+
+    public static final int POWER_X = Integer.numberOfTrailingZeros(SIZE_X);
+    public static final int POWER_Y = Integer.numberOfTrailingZeros(SIZE_Y);
+    public static final int POWER_Z = Integer.numberOfTrailingZeros(SIZE_Z);
+
+    public static final byte MAX_LIGHT = 0x0f; // max light for a light source 0-15
+    public static final byte MAX_SUNLIGHT = 0x0f; // max sunlight for sunlight bounded 0-15
+    public static final byte MAX_SUNLIGHT_REGEN = 63;
+    public static final byte SUNLIGHT_REGEN_THRESHOLD = 48;
+
+    public static final Vector3ic CHUNK_POWER = new Vector3i(POWER_X, POWER_Y, POWER_Z);
+    public static final Vector3ic CHUNK_SIZE = new Vector3i(SIZE_X, SIZE_Y, SIZE_Z);
+    public static final Vector3ic INNER_CHUNK_POS_FILTER = new org.joml.Vector3i(INNER_CHUNK_POS_FILTER_X, INNER_CHUNK_POS_FILTER_Y, INNER_CHUNK_POS_FILTER_Z);
+    public static final BlockRegion CHUNK_REGION = BlockRegions.createFromMinAndSize(new Vector3i(), CHUNK_SIZE);
+
+    public static final Vector3ic LOCAL_REGION_EXTENTS = new Vector3i(1, 1, 1);
+
+    private Chunks() {
+    }
+
+    /**
+     * Returns the chunk position of a given coordinate.
```

Comment:
The chunk position is a single integer - what does this position mean?

Reply:
And should this method ever be used directly (with a different `chunkPower` than the one from the constants)?